README.md

```diff
@@ -31,6 +31,26 @@ jsc.assert(boolFnAppliedThrice);
 
 ## Documentation
 
+### Usage with [Jest](https://jestjs.io)
+
+Using jsverify with Jest is very easy, you can either use `jsc.property` out of the box:
+
+```js
+describe("sort", function () {
+  jsc.property("(b && b) === b", jsc.bool, b => (b && b) === b);
+});
+```
+
+Or you can choose to use the helper, which exports an `itHolds` helper function. And you can use it like this:
+
+```js
+describe("a simple jest test suite", function () {
+  itHolds("(b && b) === b", jsc.bool, function (b) {
+    expect(b && b).toBe(b);
```

Comment:
I don't understand, isn't it's simpler to write

```js
describe("a simple jest test suite", function () {
  itHolds("(b && b) === b", jsc.bool, function (b) {
    expect(b && b).toBe(b);
    return true; // if you forget this your test will fail, so it's not a silent error
  });
});
```

---

Does jest support asynchronous tests (returning promise or calling callback), how these work?

Can't `itHolds` be implemented using `jsc.property`?

Reply:
@JayAndCatchFire maybe you can answer this?

Reply:
Thanks for helping out on this @jayandcatchfire

Reply:
Okay, I played a bit with `jsc.property`.  The reason I originally avoided it was because I found the documentation a bit ambiguous.  Now that I've read its source and played with it, it appears that the only important difference between my helper and `jsc.property` is that mine returns `true` at the end so the test doesn't fail even when nothing throws in the test function.

I just rewrote my function as the following and it appears to work in all the existing cases I've tried.

```typescript
function itHolds(...args: any[]) {
  const initial = args.slice(0, -1);
  const prop = args[args.length - 1];

  jsc.property(...initial, (...args: any[]) => {
    prop(...args);

    return true;
  });
}
```

Of course, this was quick and dirty.  I haven't accounted for any of the types here, which is bad form.

Reply:
Here's my final version that retains proper typing:

```typescript
import * as jsc from 'jsverify';

/* tslint:disable:max-line-length */
export function itHolds<A>(description: String, arb1: jsc.Arbitrary<A>, prop: (t: A) => jsc.Property<A>): any;
export function itHolds<A, B>(description: String, arb1: jsc.Arbitrary<A>, arb2: jsc.Arbitrary<B>, prop: (t: A, u: B) => jsc.Property<any>): any;
export function itHolds<A, B, C>(description: String, arb1: jsc.Arbitrary<A>, arb2: jsc.Arbitrary<B>, arb3: jsc.Arbitrary<C>, prop: (t: A, u: B, v: C) => jsc.Property<any>): any;
export function itHolds<A, B, C, D>(description: String, arb1: jsc.Arbitrary<A>, arb2: jsc.Arbitrary<B>, arb3: jsc.Arbitrary<C>, arb4: jsc.Arbitrary<D>, prop: (t: A, u: B, v: C, w: D) => jsc.Property<any>): any;
export function itHolds<A, B, C, D, E>(description: String, arb1: jsc.Arbitrary<A>, arb2: jsc.Arbitrary<B>, arb3: jsc.Arbitrary<C>, arb4: jsc.Arbitrary<D>, arb5: jsc.Arbitrary<E>, prop: (t: A, u: B, v: C, w: D, e: E) => jsc.Property<any>): any;
export function itHolds<A, B, C, D, E, F>(description: String, arb1: jsc.Arbitrary<A>, arb2: jsc.Arbitrary<B>, arb3: jsc.Arbitrary<C>, arb4: jsc.Arbitrary<D>, arb5: jsc.Arbitrary<E>, arb6: jsc.Arbitrary<F>, prop: (t: A, u: B, v: C, w: D, e: E, a: F) => jsc.Property<any>): any;
export function itHolds<A, B, C, D, E, F, G>(description: String, arb1: jsc.Arbitrary<A>, arb2: jsc.Arbitrary<B>, arb3: jsc.Arbitrary<C>, arb4: jsc.Arbitrary<D>, arb5: jsc.Arbitrary<E>, arb6: jsc.Arbitrary<F>, arb7: jsc.Arbitrary<G>, prop: (t: A, u: B, v: C, w: D, e: E, a: F, b: G) => jsc.Property<any>): any;
export function itHolds<A, B, C, D, E, F, G, H>(description: String, arb1: jsc.Arbitrary<A>, arb2: jsc.Arbitrary<B>, arb3: jsc.Arbitrary<C>, arb4: jsc.Arbitrary<D>, arb5: jsc.Arbitrary<E>, arb6: jsc.Arbitrary<F>, arb7: jsc.Arbitrary<G>, arb8: jsc.Arbitrary<H>, prop: (t: A, u: B, v: C, w: D, e: E, a: F, b: G, c: H) => jsc.Property<any>): any;
export function itHolds<A, B, C, D, E, F, G, H, I>(description: String, arb1: jsc.Arbitrary<A>, arb2: jsc.Arbitrary<B>, arb3: jsc.Arbitrary<C>, arb4: jsc.Arbitrary<D>, arb5: jsc.Arbitrary<E>, arb6: jsc.Arbitrary<F>, arb7: jsc.Arbitrary<G>, arb8: jsc.Arbitrary<H>, arb9: jsc.Arbitrary<I>, prop: (t: A, u: B, v: C, w: D, e: E, a: F, b: G, c: H, d: I) => jsc.Property<any>): any;
export function itHolds<A, B, C, D, E, F, G, H, I, J>(description: String, arb1: jsc.Arbitrary<A>, arb2: jsc.Arbitrary<B>, arb3: jsc.Arbitrary<C>, arb4: jsc.Arbitrary<D>, arb5: jsc.Arbitrary<E>, arb6: jsc.Arbitrary<F>, arb7: jsc.Arbitrary<G>, arb8: jsc.Arbitrary<H>, arb9: jsc.Arbitrary<I>, arb10: jsc.Arbitrary<J>, prop: (t: A, u: B, v: C, w: D, e: E, a: F, b: G, c: H, d: I, f: J) => jsc.Property<any>): any;
/* tslint:enable:max-line-length */

/**
 * wrapper function for using jsverify with BDD syntax
 */
export function itHolds(...args: any[]): any {
  const initial = args.slice(0, -1);
  const prop = args[args.length - 1];

  jsc.property(...initial, (...args: any[]): boolean => {
    prop(...args);

    return true;
  });
}
```

I still don't know if/how this would support async tests.